astropy/modeling/core.py

```diff
@@ -2236,279 +2296,103 @@ def _join_operator(f, g):
-
-
-class _CompoundModelMeta(_ModelMeta):
-    _tree = None
-    _submodels = None
-    _submodel_names = None
-    _nextid = 0
-
-    _param_names = None
-    # _param_map is a mapping of the compound model's generated param names to
-    # the parameters of submodels they are associated with.  The values in this
-    # mapping are (idx, name) tuples were idx is the index of the submodel this
-    # parameter is associated with, and name is the same parameter's name on
-    # the submodel
-    # In principle this will allow compound models to give entirely new names
-    # to parameters that don't have to be the same as their original names on
-    # the submodels, but right now that isn't taken advantage of
-    _param_map = None
-
-    _slice_offset = 0
-    # When taking slices of a compound model, this keeps track of how offset
-    # the first model in the slice is from the first model in the original
-    # compound model it was taken from
-
-    # This just inverts _param_map, swapping keys with values.  This is also
-    # useful to have.
-    _param_map_inverse = None
-    _fittable = None
-
-    _evaluate = None
-
-    def __getitem__(cls, index):
-        index = cls._normalize_index(index)
-
-        if isinstance(index, (int, np.integer)):
-            return cls._get_submodels()[index]
-        else:
-            return cls._get_slice(index.start, index.stop)
-
-    def __getattr__(cls, attr):
-        # Make sure the _tree attribute is set; otherwise we are not looking up
-        # an attribute on a concrete compound model class and should just raise
-        # the AttributeError
-        if cls._tree is not None and attr in cls.param_names:
-            cls._init_param_descriptors()
-            return getattr(cls, attr)
-
-        raise AttributeError(attr)
 
-    def __repr__(cls):
-        if cls._tree is None:
-            # This case is mostly for debugging purposes
-            return cls._format_cls_repr()
-
-        expression = cls._format_expression()
-        components = cls._format_components()
-        keywords = [
-            ('Expression', expression),
-            ('Components', '\n' + indent(components))
-        ]
+def _add_special_operator(sop_name, sop):
+    SPECIAL_OPERATORS[sop_name] = sop
 
-        return cls._format_cls_repr(keywords=keywords)
 
-    def __dir__(cls):
-        """
-        Returns a list of attributes defined on a compound model, including
-        all of its parameters.
-        """
+"""
+This module provides an alternate implementation of compound models that
+is lighter weight than the default implementation.
 
-        basedir = super().__dir__()
+Using this alternate version of compound models requires calling a function
+in core to make this one the default. If this is used, *is is higly recommended
+that the mode be set back to the default at the end of the code constructing
+compound models so that other code depending on the default behavior is
+not affected!*
```

Comment:
This is no longer true, right? The default implementation is the light implementation?